### PR TITLE
[CARBONDATA-2479] Multiple issue fixes in SDK writer and external table flow

### PR DESCRIPTION
**Pull request by ajantha-bhat**
[CARBONDATA-2479] Multiple issues:
    fixed external table path display
    fixed default value for array in AVRO
    fixed NPE when delete folder before the second select query
    fixed: avro float value precision change issue

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?NA
 
 - [ ] Any backward compatibility impacted?NA
 
 - [ ] Document update required?NA

 - [ ] Testing done
yes updated the UT

 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4718/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4912/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5874/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5881/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4727/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4940/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4751/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5905/

**Comment by kunal642** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4760/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5915/

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5925/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4770/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4956/

**Comment by ajantha-bhat** [comment]
@ravipesala : PR is ready

File: store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java [hunk 1]
```diff
@@ -540,4 +543,19 @@ private CarbonLoadModel buildLoadModel(CarbonTable table, long UUID, String task
...
+
+  /**
+   * can set the sortScope to NO_SORT
+   *
+   * @param sortScope can be set to "NO_SORT" to avoid sorting during load.
+   * by default sort_scope is "LOCAL_SORT"
+   * @return updated CarbonWriterBuilder
+   */
+  public CarbonWriterBuilder sortScope(String sortScope) {
+    if (!sortScope.equalsIgnoreCase("NO_SORT")) {
```
[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java]: Removed it and also avro Timestamp issue

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**